contract/src/msgs/staking/execute/withdraw.rs

```diff
@@ -32,9 +32,13 @@ impl ExecuteHandler for execute::withdraw::Execute {
             state::STAKERS.update(deps.storage, public_key, &executor)?;
         }
 
-        // send the tokens back to the executor
+        // send the tokens back to the specified address
+        let addr = deps
+            .api
+            .addr_validate(&self.withdraw_address)
+            .map_err(|_| ContractError::InvalidAddress(self.withdraw_address))?;
```

Comment:
Late to my own party, but I was thinking if maybe we should require the sender of the message to be the address that the user is trying to withdraw to? To prevent accidentally sending tokens to an address you don't have the private key for?

Reply:
That makes sense to me!